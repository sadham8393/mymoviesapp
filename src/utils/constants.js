// URl constants

export const MOVIEDB_PATH = "https://api.themoviedb.org/3/";
export const REACT_MOVIEDB_API_KEY  = "?api_key=0e89680c8b3fa0f259b53bbcc8f6091d";
export const MOVIEDB_IMAGE_APPEND_URL = "https://image.tmdb.org/t/p/w500/";
export const TRENDING_DAY_URL = "trending/all/day";


export const NO_IMAGE_URl = "https://s3-ap-southeast-1.amazonaws.com/upcode/static/default-image.jpg";

//Common Constants
export const INPUT_TYPE_TEXT = "text";
export const INPUT_TYPE_SELECT = "select";
export const NOT_AVAILABLE = "Not Available";


//Action Constants
export const FETCH_TRENDING_MOVIES = "FETCH_TRENDING_MOVIES";
export const FETCH_TV_MOVIES = "FETCH_TV_MOVIES";
export const DELETE_TV_MOVIES = "DELETE_TV_MOVIES";
export const MOVIES_FETCHED_SUCCESS = "MOVIES_FETCHED_SUCCESS";
export const MOVIES_FETCHED_ERROR = "MOVIES_FETCHED_ERROR";
export const DELETE_MOVIE_SUCCESS = "DELETE_MOVIE_SUCCESS";
export const DELETE_MOVIE_FAILURE = "DELETE_MOVIE_FAILURE";

export const MOVIES_FETCHED_NO_RESULTS = "MOVIES_FETCHED_NO_RESULTS";

export const RESET_RESPONSE = "RESET_RESPONSE";

export const NO_RESULTS_FOUND = "No results found for the given search input";

export const SERVER_UNAVAILABE = "Server unavailable. Please try after sometime"



