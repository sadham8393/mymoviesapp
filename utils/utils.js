import configureStore from 'redux-mock-store'

export const findByTestClass = (component, className) => {
    const wrapper = component.find(`${className}`);
    return wrapper;
};

export const findByComponent = (component, innerComponent) => {
    const wrapper = component.find(innerComponent);
    return wrapper;
};

export const testStore = (initialState) => {
    let store;
    const mockStore = configureStore();
    store = mockStore(initialState)
    return store;
};