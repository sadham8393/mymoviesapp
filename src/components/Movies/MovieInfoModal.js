import React from 'react';
import { Modal, ModalHeader, ModalBody, ModalFooter, Button} from 'reactstrap';
import { convertDate } from '../../utils/utils';
import PropTypes from 'prop-types';

const MovieInfoModal = (props) =>{
    let { title, name, overview, first_air_date, release_date} = props.movie;
    return (
        <Modal isOpen= {props.viewInfoModalOpen} >
                <ModalHeader close={<button className="close" onClick={props.closeInfoModal}>&times;</button>} > {title ? title : name} </ModalHeader>
                <ModalBody>
                    <div className ="container movies-container">
                        <p> <span className= "label"> Release Date : </span> {release_date ? convertDate(release_date) : convertDate(first_air_date)}</p>
                        <h6>Overview : </h6>
                        <p>{overview}</p>
                    </div>
                </ModalBody>
                <ModalFooter>
                    <Button color="primary" onClick={props.closeInfoModal}>OK</Button>
                </ModalFooter>
        </Modal>
    )
}

MovieInfoModal.propTypes = { 
    title: PropTypes.string, 
    name: PropTypes.string,
    closeInfoModal: PropTypes.func,
    overview : PropTypes.object,
    first_air_date : PropTypes.string,
    release_date : PropTypes.string 
} 

export default MovieInfoModal;