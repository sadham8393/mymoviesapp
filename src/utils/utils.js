import { NOT_AVAILABLE } from './constants';

/**
 * Convert date format from "yyyy-mm-dd" to "dd/mm/yy"
 */

export function convertDate(date){
    if(date){
        return date.toString().split("-").reverse().join("/");
    }
    return NOT_AVAILABLE;
}