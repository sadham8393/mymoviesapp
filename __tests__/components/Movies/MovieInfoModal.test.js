import React from 'react';
import { shallow, mount } from 'enzyme';
import EnzymeToJson from 'enzyme-to-json';
import { findByComponent } from '../../../utils/utils';
import MovieInfoModal from '../../../src/components/Movies/MovieInfoModal';
import { Modal, ModalHeader, ModalBody, ModalFooter, Button} from 'reactstrap';

describe('<MovieInfoModal /> component', () => { 
    let wrapper;
    let testProps;
    let mockFunc;
    beforeEach(() => {
        mockFunc = jest.fn();
        testProps = {
            movie : {
                title : "Avengers", 
                name : "Avengers", 
                overview : "Avengers movie info", 
                first_air_date : "03-12-2019" , 
                release_date :"03-12-2019"
            },            
            closeInfoModal : mockFunc,
            viewInfoModalOpen : true
        };
        wrapper = shallow(<MovieInfoModal {...testProps} />);
    });

    it('should render correctly', () => {
        const component = mount(<MovieInfoModal {...testProps} />);
        expect(EnzymeToJson(component)).toMatchSnapshot();
    });

    it('should render Modal correctly and to be 1', () => {
        const modal = findByComponent(wrapper, Modal);
        expect(modal.length).toBe(1);
    });

    it('should render ModalHeader correctly and to be 1', () => {
        const modal = findByComponent(wrapper, ModalHeader);
        expect(modal.length).toBe(1);
    });

    it('should render ModalBody correctly and to be 1', () => {
        const modal = findByComponent(wrapper, ModalBody);
        expect(modal.length).toBe(1);
    });

    it('should render ModalFooter correctly and to be 1', () => {
        const modal = findByComponent(wrapper, ModalFooter);
        expect(modal.length).toBe(1);
    });

    it('should render Button correctly and to be 1', () => {
        const modal = findByComponent(wrapper, Button);
        expect(modal.length).toBe(1);
    });

    it('button should emit callback on click event', () => {
        const button = findByComponent(wrapper, Button);
        button.simulate('click');
        const callback = mockFunc.mock.calls.length;
        expect(callback).toBe(1);
    });
});


