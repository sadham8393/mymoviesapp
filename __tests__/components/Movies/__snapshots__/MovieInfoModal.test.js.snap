// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieInfoModal /> component should render correctly 1`] = `
<MovieInfoModal
  closeInfoModal={[MockFunction]}
  movie={
    Object {
      "first_air_date": "03-12-2019",
      "name": "Avengers",
      "overview": "Avengers movie info",
      "release_date": "03-12-2019",
      "title": "Avengers",
    }
  }
  viewInfoModalOpen={true}
>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransition={
      Object {
        "mountOnEnter": true,
        "timeout": 150,
      }
    }
    centered={false}
    fade={true}
    isOpen={true}
    keyboard={true}
    modalTransition={
      Object {
        "timeout": 300,
      }
    }
    onClosed={[Function]}
    onOpened={[Function]}
    returnFocusAfterClose={true}
    role="dialog"
    scrollable={false}
    unmountOnClose={true}
    zIndex={1050}
  >
    <Portal
      node={
        <div
          style="position: relative; z-index: 1050; display: block;"
          tabindex="-1"
        >
          <div
            class=""
          >
            <div
              class="modal fade"
              role="dialog"
              style="display: block;"
              tabindex="-1"
            >
              <div
                class="modal-dialog"
                role="document"
              >
                <div
                  class="modal-content"
                >
                  <div
                    class="modal-header"
                  >
                    <h5
                      class="modal-title"
                    >
                       
                      Avengers
                       
                    </h5>
                    <button
                      class="close"
                    >
                      ×
                    </button>
                  </div>
                  <div
                    class="modal-body"
                  >
                    <div
                      class="container movies-container"
                    >
                      <p>
                         
                        <span
                          class="label"
                        >
                           Release Date : 
                        </span>
                         
                        2019/12/03
                      </p>
                      <h6>
                        Overview : 
                      </h6>
                      <p>
                        Avengers movie info
                      </p>
                    </div>
                  </div>
                  <div
                    class="modal-footer"
                  >
                    <button
                      class="btn btn-primary"
                      type="button"
                    >
                      OK
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="modal-backdrop fade"
            />
          </div>
        </div>
      }
    >
      <Portal
        containerInfo={
          <div
            style="position: relative; z-index: 1050; display: block;"
            tabindex="-1"
          >
            <div
              class=""
            >
              <div
                class="modal fade"
                role="dialog"
                style="display: block;"
                tabindex="-1"
              >
                <div
                  class="modal-dialog"
                  role="document"
                >
                  <div
                    class="modal-content"
                  >
                    <div
                      class="modal-header"
                    >
                      <h5
                        class="modal-title"
                      >
                         
                        Avengers
                         
                      </h5>
                      <button
                        class="close"
                      >
                        ×
                      </button>
                    </div>
                    <div
                      class="modal-body"
                    >
                      <div
                        class="container movies-container"
                      >
                        <p>
                           
                          <span
                            class="label"
                          >
                             Release Date : 
                          </span>
                           
                          2019/12/03
                        </p>
                        <h6>
                          Overview : 
                        </h6>
                        <p>
                          Avengers movie info
                        </p>
                      </div>
                    </div>
                    <div
                      class="modal-footer"
                    >
                      <button
                        class="btn btn-primary"
                        type="button"
                      >
                        OK
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="modal-backdrop fade"
              />
            </div>
          </div>
        }
      >
        <div
          className=""
        >
          <Fade
            appear={true}
            baseClass="fade"
            baseClassActive="show"
            className="modal"
            enter={true}
            exit={true}
            in={true}
            mountOnEnter={false}
            onClick={[Function]}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            role="dialog"
            style={
              Object {
                "display": "block",
              }
            }
            tabIndex="-1"
            tag="div"
            timeout={300}
            unmountOnExit={false}
          >
            <Transition
              appear={true}
              enter={true}
              exit={true}
              in={true}
              mountOnEnter={false}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={300}
              unmountOnExit={false}
            >
              <div
                className="modal fade"
                onClick={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                role="dialog"
                style={
                  Object {
                    "display": "block",
                  }
                }
                tabIndex="-1"
              >
                <div
                  className="modal-dialog"
                  role="document"
                >
                  <div
                    className="modal-content"
                  >
                    <ModalHeader
                      charCode={215}
                      close={
                        <button
                          className="close"
                          onClick={[MockFunction]}
                        >
                          ×
                        </button>
                      }
                      closeAriaLabel="Close"
                      tag="h5"
                      wrapTag="div"
                    >
                      <div
                        className="modal-header"
                      >
                        <h5
                          className="modal-title"
                        >
                           
                          Avengers
                           
                        </h5>
                        <button
                          className="close"
                          onClick={[MockFunction]}
                        >
                          ×
                        </button>
                      </div>
                    </ModalHeader>
                    <ModalBody
                      tag="div"
                    >
                      <div
                        className="modal-body"
                      >
                        <div
                          className="container movies-container"
                        >
                          <p>
                             
                            <span
                              className="label"
                            >
                               Release Date : 
                            </span>
                             
                            2019/12/03
                          </p>
                          <h6>
                            Overview : 
                          </h6>
                          <p>
                            Avengers movie info
                          </p>
                        </div>
                      </div>
                    </ModalBody>
                    <ModalFooter
                      tag="div"
                    >
                      <div
                        className="modal-footer"
                      >
                        <Button
                          color="primary"
                          onClick={[MockFunction]}
                          tag="button"
                        >
                          <button
                            aria-label={null}
                            className="btn btn-primary"
                            onClick={[Function]}
                            type="button"
                          >
                            OK
                          </button>
                        </Button>
                      </div>
                    </ModalFooter>
                  </div>
                </div>
              </div>
            </Transition>
          </Fade>
          <Fade
            appear={true}
            baseClass="fade"
            baseClassActive="show"
            className="modal-backdrop"
            enter={true}
            exit={true}
            in={true}
            mountOnEnter={true}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            tag="div"
            timeout={150}
            unmountOnExit={false}
          >
            <Transition
              appear={true}
              enter={true}
              exit={true}
              in={true}
              mountOnEnter={true}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={150}
              unmountOnExit={false}
            >
              <div
                className="modal-backdrop fade"
              />
            </Transition>
          </Fade>
        </div>
      </Portal>
    </Portal>
  </Modal>
</MovieInfoModal>
`;
